File: Rado/part365/src/app/model/car/car.ts
import { ApprovedType } from "@model/enum/approveType.enum";

export class Car {
    carId?           : number;
    bus?             : number;
    modelId?         : number; 
    modificationId?  : number;
    year?            : number;
    vin?             : string;
    regNumber?       : string;
    description?     : string;
    price?           : number;
    powerkWh?        : number;
    powerBHP?        : number;
    engineType?      : number;
    engineModel?     : string;
    millage?          : number;
    gearboxType?     : number;
    modifiedTime?    : number;
    sellerPhone?     : string;
    sellerPhone2?    : string;
    sellerViber?     : string;
    sellerWebPage?   : string;
    regionId?        : number;
    userId?          : number;
    mainImageId?     : number;
    mainPicture?     : string;
    approved?        : ApprovedType;
}

File: Models/Models/Car.cs
using Rado;
using Rado.Models;
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace Models.Models
{
    public class Car : Item
    {
        protected List<ImageDataClass>? images_ = new List<ImageDataClass>();
        [Required]
        public long CarId { get; set; }
        public int? ModelId { get; set; }
        public int? ModificationId { get; set; }
        public int Year { get; set; }
        public string? Vin { get; set; }
        [Required]
        public string RegNumber { get; set; }
        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingDefault)]
        public int PowerkWh { get; set; }
        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingDefault)]
        public int PowerBHP { get; set; }
        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingDefault)]
        public int EngineType { get; set; }
        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingDefault)]
        public string? EngineModel { get; set; }
        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingDefault)]
        public int Millage { get; set; }
        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingDefault)]
        public int GearboxType { get; set; }
        public int Bus { get; set; }
        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public Part[]? Parts { get; set; }
    }
}
